Sources/Networking.swift

```diff
@@ -154,37 +152,60 @@ public class Networking {
      - parameter username: The username to be used.
      - parameter password: The password to be used.
      */
+    @available(*, deprecated: 2.2.0, message: "Use `var basicAuthenticationHeaderField` instead.")
     public func authenticate(username: String, password: String) {
-        let credentialsString = "\(username):\(password)"
-        if let credentialsData = credentialsString.data(using: .utf8) {
-            let base64Credentials = credentialsData.base64EncodedString(options: [])
-            let authString = "Basic \(base64Credentials)"
-
-            let config = self.sessionConfiguration()
-            config.httpAdditionalHeaders = [self.authorizationHeaderKey as AnyHashable: authString]
-            self.session = URLSession(configuration: config)
-        }
+        self.basicAuthenticationHeaderField = (username, password)
     }
 
     /** 
      Authenticates using a Bearer token, sets the Authorization header to "Bearer \(token)".
      - parameter token: The token to be used.
      */
+    @available(*, deprecated: 2.2.0, message: "Use `var authenticationHeaderFieldValue` with \"Bearer \\(token)\" as value instead.")
```

Comment:
Why is this being set directly? The methods I had before worked better than this. 😢 Less error prone, more clarity, no need to rely on tuples...